### PR TITLE
Feature/load pets by guardian

### PR DESCRIPTION
**Pull request by andersonnaz**
create LoadPets use case
create LoadPets Controller
create route /pet with method GET to return a list of pets by Guardian
create tests for all features like usecase, controller and route
update documentation with new route.

### CODE REVIEW

**Comment by matheusgondra** [comment]
Está faltando os testes do `LoadPetsController`

File: tests/src/infra/repos/postgresql/pet-repository.spec.ts [hunk 1]
```diff
@@ -12,91 +12,158 @@ const makeSut = (): PetRepository => {
...
+        gender: PetGender.MALE,
+        breedId: breed.id,
+        breedAlias: 'any_breed_alias',
+        sizeId: size.id,
+        castrated: false
       }
+
+      await sut.add(data)
+      const result = await sut.load(data.guardianId)
+      expect(result).toBeTruthy()
```
Comment (matheusgondra) [inline tests/src/infra/repos/postgresql/pet-repository.spec.ts]: Esse expect não é necessário já que está testando com o `toEqual()` na linha abaixo

File: tests/src/application/controllers/pets/load-pets.spec.ts
Comment (matheusgondra) [inline tests/src/application/controllers/pets/load-pets.spec.ts:1]: Create a test that checks the `LoadPets` call inside the controller

**Review by matheusgondra — CHANGES_REQUESTED** [review]

**Review by matheusgondra — COMMENTED** [review]
I think we should test the return case when there are no pets registered. Test an empty list return or return a 204 status from noContent

**Review by rodrigofmeneses — DISMISSED** [review]
LGFM

**Review by rodrigofmeneses — APPROVED** [review]

**Review by matheusgondra — APPROVED** [review]

**Pull request merged**